sdk/metric/periodic_reader.go

```diff
@@ -219,6 +219,9 @@ func (r *periodicReader) aggregation(kind InstrumentKind) aggregation.Aggregatio
 // collectAndExport gather all metric data related to the periodicReader r from
 // the SDK and exports it with r's exporter.
 func (r *periodicReader) collectAndExport(ctx context.Context) error {
+	ctx, cancel := context.WithTimeout(ctx, r.timeout)
+	defer cancel()
```

Comment:
I know this isn't tested. But I'm getting [weird failures](https://github.com/open-telemetry/opentelemetry-go/actions/runs/5278439315/jobs/9547778642?pr=4221) in some environments.
I intend to look at improving the test later on, but since this wasn't tested at all before, I figured it'd be simpler to split both changes.